File: app/src/main/java/com/expensio/domain/model/RecurringExpense.kt
package com.expensio.domain.model

data class RecurringExpense(
    val id: String,
    val groupId: String,
    val title: String,
    val amount: Double,
    val category: String,
    val paidByUserId: String?,
    val paidByGuestId: String?,
    val paidByName: String,
    val splitType: String,
    val frequency: String,
    val nextDueDate: String,
    val isActive: Boolean,
    val createdAt: String,
)

enum class Frequency(val label: String) {
    DAILY("Daily"),
    WEEKLY("Weekly"),
    MONTHLY("Monthly"),
    YEARLY("Yearly"),
}

File: backend/app/schemas/recurring_expense.py
import uuid
from datetime import date, datetime
from decimal import Decimal
from enum import Enum
from typing import Optional

from pydantic import BaseModel

from app.schemas.expense import SplitInput


class FrequencyEnum(str, Enum):
    DAILY = "DAILY"
    WEEKLY = "WEEKLY"
    MONTHLY = "MONTHLY"
    YEARLY = "YEARLY"


class RecurringExpenseCreate(BaseModel):
    title: str
    amount: Decimal
    category: str
    paid_by_user_id: Optional[uuid.UUID] = None
    paid_by_guest_id: Optional[uuid.UUID] = None
    split_type: str = "EQUAL"
    splits: list[SplitInput] = []
    frequency: FrequencyEnum
    start_date: date


class RecurringExpenseResponse(BaseModel):
    id: uuid.UUID
    group_id: uuid.UUID
    title: str
    amount: Decimal
    category: str
    paid_by_user_id: Optional[uuid.UUID]
    paid_by_guest_id: Optional[uuid.UUID]
    paid_by_name: str
    split_type: str
    frequency: str
    next_due_date: date
    is_active: bool
    created_at: datetime
